security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,283 @@
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+  1.  Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program.
+      - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+      - [Proof Test](https://github.com/ethereum-optimism/optimism/blob/723f33f181dc0e9946599adcaa2ba55489960a85/op-e2e/actions/proofs/fixtures.go#L50)
+        - op-program and other fault proof program tests uses this fixture to ensure transactions utilizing these precompiles succeed.
+- **Detection:** The `vm-runner` computation time would **increase** significantly, or **panic** due to OOM and trigger an alert through our current set of monitoring.
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute
+  BLS precompiles in sufficient time.
+
+### FM6: BLS Precompiles could fail to execute in the FP program
+
+- **Description:** If implemented in the FP program, BLS precompiles could fail to execute. This could cause certain blocks to be unprovable.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+
+  1.  Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program. This means it should
+      have the same performance characteristics as existing crypto precompiles like `ecrecover`. - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes) - [Proof Test](https://github.com/ethereum-optimism/optimism/blob/723f33f181dc0e9946599adcaa2ba55489960a85/op-e2e/actions/proofs/fixtures.go#L50) - op-program and other fault proof program tests uses this fixture to ensure transactions utilizing these precompiles succeed.
+
+- **Detection:** op-program trace runner failure (possibly triggering an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute BLS precompiles in sufficient time.
+
+### FM7: Early fork if batches containing EIP-7702 transactions could be posted before Pectra
+
+- **Description:** If batches containing EIP-7702 transactions are posted and accepted early, the batcher could fork any nodes that upgrade early. A malicious batcher could cause this to happen if we don't have a check to ensure type 4 (SetCode) txs can only appear in a batch after Isthmus is active.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+  1. Spec disallows type 4 transactions before Isthmus
+  1. End-to-end test
+     - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
```

Comment:
I think we should be linking to this test here? https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/actions/proofs/isthmus_setcode_tx_test.go 

Also, I'm not sure we have coverage for the unsafe chain, i.e. a check that the txpool will filter out set code txs if Isthmus is not active.